### PR TITLE
Add build status and coverage percentage in README.md

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
This PR adds latest build statuses and code coverage percentages from public pipelines of each component - azfile, azblob and azdatalake.
The output will be similar to the following - 
image. 

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azblob/README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Azure Blob Storage module for Go
+[![Build Status](https://dev.azure.com/azure-sdk/public/_apis/build/status/go/go%20-%20azdatalake%20-%20ci?branchName=main)](https://dev.azure.com/azure-sdk/public/_build/latest?definitionId=2854&branchName=main)
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/README.md:2]: I don't see GO reference part in the readme as mentioned in the PR description.

Comment (tanyasethi-msft) [reply sdk/storage/azblob/README.md:3]: The commit for go-reference was not checked in. Checked in now.

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**